amethyst_gltf/src/format/mesh.rs

```diff
@@ -0,0 +1,226 @@
+                        once(normal.clone())
+                            .chain(once(normal.clone()))
+                            .chain(once(normal))
+                    })
+                    .collect::<Vec<_>>()
+            });
+
+        debug!("Loading texture coordinates");
+        let tex_coord = reader
+            .read_tex_coords(0)
+            .map(|tex_coords| tex_coords.into_f32().collect::<Vec<[f32; 2]>>())
+            .unwrap_or_else(|| {
+                vec![
+                    [options.generate_tex_coords.0, options.generate_tex_coords.1];
+                    positions.len()
+                ]
+            });
+        let tex_coord: Vec<[f32; 2]> = match faces {
+            Some(ref faces) => faces
+                .iter()
+                .map(|i| flip_check(tex_coord[*i], options.flip_v_coord))
+                .collect(),
+            None => tex_coord
+                .into_iter()
+                .map(|t| flip_check(t, options.flip_v_coord))
+                .collect(),
+        };
+
+        debug!("Loading tangents");
+        let tangents = reader
+            .read_tangents()
+            .map(|tangents| match faces {
+                Some(ref faces) => {
+                    let tangents = tangents.collect::<Vec<_>>();
+                    faces
+                        .iter()
+                        .map(|i| [tangents[*i][0], tangents[*i][1], tangents[*i][2]])
+                        .collect()
+                }
+                None => tangents.map(|t| [t[0], t[1], t[2]]).collect(),
+            })
+            .unwrap_or_else(|| {
+                let f = faces
+                    .as_ref()
+                    .map(|f| f.clone())
+                    .unwrap_or_else(|| (0..positions.len()).collect::<Vec<_>>());
+                let vertices_per_face = || 3;
+                let face_count = || f.len() / 3;
+                let p = |face, vert| &positions[f[face * 3 + vert]];
+                let n = |face, vert| &normals[f[face * 3 + vert]];
+                let tx = |face, vert| &tex_coord[f[face * 3 + vert]];
+                let mut tangents: Vec<(usize, [f32; 4])> = Vec::with_capacity(f.len());
+                {
+                    let mut set_tangent = |face, vert, tangent| {
+                        let index = face * 3 + vert;
+                        if let Err(pos) = tangents.binary_search_by(|probe| probe.0.cmp(&index)) {
+                            tangents.insert(pos, (index, tangent));
+                        }
+                    };
+                    mikktspace::generate_tangents(
+                        &vertices_per_face,
+                        &face_count,
+                        &p,
+                        &n,
+                        &tx,
+                        &mut set_tangent,
+                    );
+                }
+
+                tangents.iter().map(|(_, t)| [t[0], t[1], t[2]]).collect()
+            });
+
+        debug!("Loading bounding box");
```

Comment:
Those are many debug calls, are you sure we need them all / they should log at debug?